### PR TITLE
Turnitin LTI

### PR DESCRIPTION
**Pull request by buckett**

### CODE REVIEW

File: assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -1514,6 +1514,23 @@ protected String build_student_view_submission_context(VelocityPortlet portlet,
 				context.put("student",student);
 			}
 		}
+		
+		//TODO move upwards inside some checking
+		Site site = null;
+		try {
+			site = SiteService.getSite(contextString);
```
Comment (buckett) [inline assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Loading the site doesn't look to be the pattern `buildMainPanelContext` puts the site in the context for most templates and it should probably so it for this one.

File: assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -3663,6 +3680,22 @@ protected String build_instructor_view_assignment_context(VelocityPortlet portle
 		context.put("contentTypeImageService", state.getAttribute(STATE_CONTENT_TYPE_IMAGE_SERVICE));
 		context.put("honor_pledge_text", ServerConfigurationService.getString("assignment.honor.pledge", rb.getString("gen.honple2")));
 		
+		//TODO add checkings if assignment uses tii
+		Site s = null;
```
Comment (buckett) [inline assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: The site is already in the `context` under the key `"site"`

**Comment by buckett** [comment]
This doesn't currently merge with our stable branch.

**Comment by bbbbgarcia** [comment]
Any clues of the reason why? What can/should we do?

**Comment by buckett** [comment]
Either rebase against sakai-10.x or merge the changes form sakai-10.x into your branch.

It's probably worth squashing some of the commits down first so that there aren't as many.

**Comment by buckett** [comment]
This still isn't merging against the sakai-10.x branch.

**Comment by bbbbgarcia** [comment]
Didn't make any changes on that regard yet, but I think it could have to do
with the Oxford only code? Are you planning to merge to the 10.x soon?

2016-03-09 15:16 GMT+01:00 Matthew Buckett notifications@github.com:

**Comment by buckett** [comment]
Yep, we're looking to get this all merged very soon.

On 9 March 2016 at 14:30, bgarciaentornos notifications@github.com wrote:

> Didn't make any changes on that regard yet, but I think it could have to do
> with the Oxford only code? Are you planning to merge to the 10.x soon?
> 
> 2016-03-09 15:16 GMT+01:00 Matthew Buckett notifications@github.com:
> 
> > This still isn't merging against the sakai-10.x branch.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/ox-it/wl-assignment/pull/24#issuecomment-194312447.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ox-it/wl-assignment/pull/24#issuecomment-194319857.

## 

  Matthew Buckett, VLE Developer, IT Services, University of Oxford

**Comment by bbbbgarcia** [comment]
Oh, I thought you meant the official Sakai 10.x branch.

2016-03-09 15:43 GMT+01:00 Matthew Buckett notifications@github.com:

**Comment by buckett** [comment]
This still doesn't merge against our Sakai 10.x branch.

**Comment by bbbbgarcia** [comment]
I know, haven't got to that yet. Hopefully by tomorrow.

File: assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java [hunk 3]
```diff
@@ -5819,6 +5858,360 @@ public boolean willArchiveMerge()
...
+			} // output resource
+
+		}
+		catch (Exception t)
+		{
+			M_log.error("Exception handling content " + t.getMessage());
+		}
+	}
+	
+	protected ArrayList<Range> parseRange(HttpServletRequest request,
```
Comment (buckett) [inline assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java:6050]: Ranged GET doesn't make any sense here does it?

File: assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java [hunk 4]
```diff
@@ -6917,6 +7369,12 @@ else if (REF_TYPE_SUBMISSION.equals(ref.getSubType()))
 	 */
 	public boolean canSubmit(String context, Assignment a)
 	{
+		// submissions are never allowed to non-electronic assignments  --bbailla2
```
Comment (buckett) [inline assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java:7372]: Indentation

**Pull request merged**